docs/SPEC.md

```diff
@@ -405,10 +405,28 @@ The default value is set using the time zone of the running process.
 ### Types
 
 A type defines a set of values and operations on those values.
-Types are never explicitly declared as part of the syntax.
+Types are never explicitly declared as part of the syntax except as part of a [builtin statement](#built-ins).
 Types are always inferred from the usage of the value.
 Type inference follows a Hindley-Milner style inference system.
 
+    polytype = monotype ["where" constraints] .
+    monotype = tvar | basic | array | record | function .
+    tvar     = "A" … "Z" .
```

Comment:
do we want to allow for `t#` format types as well in cases where there are lots of type variables? Or only use letters?

Reply:
I think we should only use letters for now. To need any more than 3 type variables will be extremely rare at best.